perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""MySQL Service Benchmarks.
+
+This is a set of benchmarks that measures performance of MySQL Databases on
+managed MySQL services.
+
+- On AWS, we will use RDS+MySQL.
+- On GCP, we will use Cloud SQL v2 (Performance Edition). As of July 2015, you
+ will need to request to whitelist your GCP project to get access to cloud sql
+ v2. Follow instructions on your GCP's project console to do that.
+- On Azure, we will use the MySQL service whenever it becomes availble.
+
+"""
+import json
+import logging
+import re
+import StringIO
+import time
+
+from perfkitbenchmarker import benchmark_spec as benchmark_spec_class
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.aws import aws_network
+
+FLAGS = flags.FLAGS
+flags.DEFINE_enum(
+    'db_instance_cores', '4', ['1', '4', '8', '16'],
+    'The number of cores to be provisioned for the DB instance.')
+
+flags.DEFINE_integer('oltp_tables_count', 4,
+                     'The number of tables used in sysbench oltp.lua tests')
+flags.DEFINE_integer('oltp_table_size', 100000,
+                     'The number of rows of each table used in the oltp tests')
+flags.DEFINE_integer('sysbench_warmup_seconds', 120,
+                     'The duration of the warmup run in which results are '
+                     'discarded.')
+flags.DEFINE_integer('sysbench_run_seconds', 480,
+                     'The duration of the actual run in which results are '
+                     'collected.')
+flags.DEFINE_integer('sysbench_thread_count', 16,
+                     'The number of test threads on the client side.')
+flags.DEFINE_integer('sysbench_latency_percentile', 99,
+                     'The latency percentile we ask sysbench to compute.')
+flags.DEFINE_integer('sysbench_report_interval', 2,
+                     'The interval we ask sysbench to report results.')
+
+BENCHMARK_INFO = {'name': 'mysql_service',
+                  'description': 'MySQL service benchmarks.',
+                  'scratch_disk': False,
+                  'num_machines': 1}
+
+# Query DB creation status once every 15 seconds
+DB_STATUS_QUERY_INTERVAL = 15
+
+# How many times we will wait for the service to create the DB
+# total wait time is therefore: "query interval * query limit"
+DB_STATUS_QUERY_LIMIT = 200
+
+AWS_PREFIX = ['aws', '--output=json']
+# Map from FLAGs.db_instance_cores to RDS DB Type
+RDS_CORE_TO_DB_CLASS_MAP = {
```

Comment:
seems like this should be in the AWS directory,  We should be abstracting away these details in this file.